laiaaa0/assignment3/Dictionary.cpp

```diff
@@ -1,47 +1,45 @@
 #include "Dictionary.h"
 
-bool Dictionary::IsPrefixOfWord(std::string prefix, std::string word){
-  if (prefix.size()>word.size()) return false;
-  else {
-    for (int i=0; i<prefix.size(); ++i){
-      if (word[i]!=prefix[i]) return false;
-    }
-    return true;
+bool Dictionary::IsPrefixOfWord(std::string prefix, std::string word) const{
+  if (prefix.size() > word.size()) return false;
+  for (int i = 0; i < prefix.size(); ++i) {
+    if (word[i] != prefix[i]) return false;
   }
+  return true;
 }
 
 
-Dictionary::Dictionary(const std::vector<std::string> &word_list){
-  for (int i = 0; i<word_list.size(); ++i){
-    std::string current_word = word_list[i];
-    InsertWord(current_word);
+Dictionary::Dictionary(const std::vector<std::string> &word_list) {
+  for (int i = 0; i < word_list.size(); ++i) {
+      InsertWord(word_list[i]);
   }
-
 }
-bool Dictionary::IsWord (const std::string & query_word){
-  if (this->dictionary_words_.find(query_word) == this->dictionary_words_.end()) return false;
-  else return (this->dictionary_words_[query_word]);
+
+bool Dictionary::IsWord(const std::string & query_word) const {
+  auto query_iterator =this->dictionary_words_.find(query_word);
+  if (query_iterator == this->dictionary_words_.end()) return false;
+
+  return query_iterator->second;
 }
 
 
-bool Dictionary::IsPrefix (const std::string & query_prefix){
-  if (this->dictionary_prefixes_.find(query_prefix) == this->dictionary_prefixes_.end()) return false;
-  else {
-    if (this->dictionary_prefixes_[query_prefix]>0) return true;
-    else return false;
-  }
+bool Dictionary::IsPrefix(const std::string & query_prefix) const {
+  auto query_iterator = this->dictionary_prefixes_.find(query_prefix);
+  if (query_iterator == this->dictionary_prefixes_.end()) return false;
 
+  if (query_iterator->second > 0) return true;
+  return false;
```

Comment:
```c++
if (query_iterator->second > 0) return true;
return false;
```
could be replaced with:
```c++
return query_iterator->second > 0;
```
or 
```c++
 if (query_iterator == this->dictionary_prefixes_.end()) return false;
 if (query_iterator->second > 0) return true;
 return false;
```
With: 
```c++
return query_iterator != this->dictionary_prefixes_.end() && query_iterator->second > 0;
```
But then readability starts to suffer.